facebook/docusaurus
# refactor(v2): combine nested theme providers into one

## Motivation

As discussed in the [comment](https://github.com/facebook/docusaurus/pull/2645#pullrequestreview-417205906), we can merge the provider for an announcement bar and tab choice into single one.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

- Try to close the announcement bar
- Try to switch tab (https://deploy-preview-2870--docusaurus-2.netlify.app/docs/markdown-features/#syncing-tab-choices)

Reload the page, changes (preferences) should be restored.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)
